sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -94,6 +101,27 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
       resolveNamespace(catalog, ns, fetchMetadata)
   }
 
+  private def resolveIdentifier(
+      nameParts: Seq[String],
+      allowTemp: Boolean,
+      columns: Seq[ColumnDefinition]): ResolvedIdentifier = {
+    val columnOutput = columns.map { col =>
+      val dataType = col.dataType match {
+        case _: CharType | _: VarcharType if !conf.preserveCharVarcharTypeInfo =>
```

Comment:
@cloud-fan I made this new change to bypass https://github.com/apache/spark/commit/38c6ef456ed67ed55e186625dcde017d85d6431f#diff-583171e935b2dc349378063a5841c5b98b30a2d57ac3743a9eccfe7bffcb8f2aR286
Does this looks good to you?